src/_pytest/_code/code.py

```diff
@@ -609,9 +609,10 @@ def match(self, regexp: "Union[str, Pattern]") -> "Literal[True]":
         If it matches `True` is returned, otherwise an `AssertionError` is raised.
         """
         __tracebackhide__ = True
-        assert re.search(
+        regextest = re.search(regexp, str(self.value))
+        assert regextest, "Regex pattern r{!r} does not match {!r}".format(
```

Comment:
I kind of liked the following suggestion from https://github.com/pytest-dev/pytest/issues/7489#issuecomment-657812669:

> If string-equal, but not regex-match, show a "Did you mean" help message

So perhaps this can be changed to something like:

```python
regextest = re.search(regexp, str(self.value))
msg = "Regex pattern r{!r} does not match {!r}."
if regexp == str(self.value):
    msg += ". Did you mean to `re.escape()` the regex?"
assert regextest, msg.format(regexp, str(self.value))
```

Reply:
In this case we should add a test which tests this message explicitly.

Reply:
😂  totally different take to that which I was going for, but unblocked me. Nice 👍 🚀

Reply:
😂 those full-stops you added... regex wildcards isn't it... hot damn can you see why this API is such a pain?

Reply:
and the parenthesis...

Reply:
Sorry not sure I'm following... 🤔

Reply:
The presence of regex characters, makes matching the exception harder, because everything is forced into a regex, that needs to be escaped in order to match. It's cool, I've got a new test-case for the new error condition and changes now. It looks like none of the existing errors covered this case.